cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -358,8 +409,34 @@ private CassandraNodeHealthCheckDetails performHealthCheck(@NotNull final Cassan
         return builder.build();
     }
 
-    private static CassandraNodeInfo buildInfo(JmxConnect jmx) throws UnknownHostException {
-        Nodetool nodetool = new Nodetool(jmx);
+    private void processAlive(ExecutorDriver driver) {
+        if (process == null)
+            return;
+        try {
+            int exitCode = process.exitValue();
+            LOGGER.error("Cassandra process terminated unexpectedly with exit code {}", exitCode);
+
+            final SlaveStatusDetails details = SlaveStatusDetails.newBuilder()
+                    .setStatusDetailsType(SlaveStatusDetails.StatusDetailsType.ERROR_DETAILS)
+                    .setSlaveErrorDetails(
+                            SlaveErrorDetails.newBuilder()
```

Comment:
Please make this status object more robust. The scheduler would probably like to be able to take the exit code into consideration when making decisions.